src/pages/RFOX/hooks/useGetUnstakingRequestCountQuery.ts

```diff
@@ -21,28 +21,30 @@ type UnstakingRequestCount = ReadContractReturnType<
 >
 type UseGetUnstakingRequestCountQueryProps<SelectData = UnstakingRequestCount> = {
   stakingAssetAccountAddress: string | undefined
+  contractAddress?: Address
   select?: (unstakingRequestCount: UnstakingRequestCount) => SelectData
 }
 const client = viemClientByNetworkId[arbitrum.id]
 
 export const useGetUnstakingRequestCountQuery = <SelectData = UnstakingRequestCount>({
   stakingAssetAccountAddress,
+  contractAddress = RFOX_PROXY_CONTRACT,
   select,
 }: UseGetUnstakingRequestCountQueryProps<SelectData>) => {
   // wagmi doesn't expose queryFn, so we reconstruct the queryKey and queryFn ourselves to leverage skipToken type safety
   const queryKey: GetUnstakingRequestCountQueryKey = useMemo(
     () => [
       'readContract',
       {
-        address: RFOX_PROXY_CONTRACT,
+        address: contractAddress,
```

Comment:
Auto review: in some hooks we pass the contract address, some other has the assetId, choose between both and stick with it

Reply:
passing `stakingAssetId` across the board now.